igvm/commands.py

```diff
@@ -543,9 +573,19 @@ def _get_vm(hostname, ignore_reserved=False):
                 dataset_obj, hv_attr, dataset_obj[hv_attr]['hostname']
             )
 
-    return VM(dataset_obj, hypervisor)
+    vm = VM(dataset_obj, hypervisor)
+    vm.lock()
+
+    try:
+        yield vm
+    finally:
+        # We can't risk commiting any other changes to the VM than unlocking.
+        # There can be changes from failed things, like setting memory.
```

Comment:
"We have to re-fetch the VM, because we can't risk..." would be clearer.

Reply:
True.